lib/TransactionKit.ts

```diff
@@ -656,111 +1106,353 @@ export class EtherspotTransactionKit implements IInitial {
+            'estimate(): Calculated cost',
+            {
+              totalGas: totalGasBigInt.toString(),
+              maxFeePerGas: maxFeePerGas.toString(),
+              cost: cost.toString(),
+            },
+            this.debugMode
+          );
+
+          log(
+            'estimate(): Single transaction estimated successfully (delegatedEoa)',
+            {
+              to: this.workingTransaction?.to,
+              cost: cost.toString(),
+              gasUsed: totalGasBigInt.toString(),
+            },
+            this.debugMode
+          );
+
+          // Success: reset error states
+          this.isEstimating = false;
+          this.containsEstimatingError = false;
+
+          // Get the current nonce for the smart account
+          const nonce = await publicClient.getTransactionCount({
+            address: delegatedEoaAccount.address,
+            blockTag: 'pending',
+          });
+
+          // Encode the call data for the transaction
+          const callData = await delegatedEoaAccount.encodeCalls([call]);
+
+          log(
+            'estimate(): Got UserOp data',
+            {
+              nonce: nonce.toString(),
+              callData: callData,
+              sender: delegatedEoaAccount.address,
+            },
+            this.debugMode
+          );
+
+          // Create an UserOp object
+          const userOp: UserOp = {
+            sender: delegatedEoaAccount.address,
+            nonce: BigInt(nonce),
+            callData: callData,
+            callGasLimit: gasEstimate.callGasLimit || BigInt(0),
+            verificationGasLimit: gasEstimate.verificationGasLimit || BigInt(0),
+            preVerificationGas: gasEstimate.preVerificationGas || BigInt(0),
+            maxFeePerGas: maxFeePerGas, // Use proper EIP-1559 maxFeePerGas
+            maxPriorityFeePerGas: maxPriorityFeePerGas, // Use proper EIP-1559 maxPriorityFeePerGas
+            paymasterData: '0x', // Paymaster not supported in delegatedEoa mode
+            signature: '0x', // Will be set during actual send
+            factory: undefined,
+            factoryData: undefined,
+            paymaster: undefined, // Paymaster not supported in delegatedEoa mode
+            paymasterVerificationGasLimit:
+              gasEstimate.paymasterVerificationGasLimit || BigInt(0),
+            paymasterPostOpGasLimit:
+              gasEstimate.paymasterPostOpGasLimit || BigInt(0),
+          };
+
+          const result = {
+            to: this.workingTransaction?.to || '',
+            value: this.workingTransaction?.value?.toString(),
+            data: this.workingTransaction?.data,
+            chainId: this.workingTransaction!.chainId,
+            cost,
+            userOp,
+            isEstimatedSuccessfully: true,
+          };
+          log(
+            'estimate(): Returning success result (delegatedEoa)',
+            result,
+            this.debugMode
+          );
+          return { ...result, ...this };
```

Comment:
did you mean to spread `this` here? Just incase it exposes quite a lot of data...